api/src/main/resources/workbench.yaml

```diff
@@ -922,6 +922,249 @@ paths:
+        - in: path
+          name: cdrVersionId
+          type: integer
+          format: int64
+          required: true
+          description: specifies which cdr version
+        - in: body
+          name: request
+          required: true
+          description: CohortAnnotationDefinition creation request body
+          schema:
+            $ref: "#/definitions/CreateCohortAnnotationDefinitionRequest"
+      responses:
+        "200":
+          description: A CohortAnnotationDefinition.
+          schema:
+            $ref: "#/definitions/CohortAnnotationDefinition"
+    get:
+      tags:
+        - cohortReview
+      description: Returns a collection of CohortAnnotationDefinition.
+      operationId: "getCohortAnnotationDefinitions"
+      parameters:
+        - in: path
+          name: workspaceNamespace
+          type: string
+          required: true
+        - in: path
+          name: workspaceId
+          type: string
+          required: true
+          description: specifies which workspace
+        - in: path
+          name: cohortId
+          type: integer
+          format: int64
+          required: true
+          description: specifies which cohort
+        - in: path
+          name: cdrVersionId
+          type: integer
+          format: int64
+          required: true
+          description: specifies which cdr version
+      responses:
+        "200":
+          description: A collection of CohortAnnotationDefinition
+          schema:
+            $ref: "#/definitions/CohortAnnotationDefinitionListResponse"
+
+  /api/v1/workspaces/{workspaceNamespace}/{workspaceId}/review/{cohortId}/{cdrVersionId}/annotationdefinitions/{annotationDefinitionId}:
+    put:
+      tags:
+        - cohortReview
+      description: modify the CohortAnnotationDefinition.
+      operationId: updateCohortAnnotationDefinition
+      parameters:
+        - in: path
+          name: workspaceNamespace
+          type: string
+          required: true
+          description: specifies which workspace namespace
+        - in: path
+          name: workspaceId
+          type: string
+          required: true
+          description: specifies which workspace
+        - in: path
+          name: cohortId
+          type: integer
+          format: int64
+          required: true
+          description: specifies which cohort
+        - in: path
+          name: cdrVersionId
+          type: integer
+          format: int64
+          required: true
+          description: specifies which cdr version
+        - in: path
+          name: annotationDefinitionId
+          type: integer
+          format: int64
+          required: true
+          description: specifies which CohortAnnotationDefinition.
+        - in: body
+          name: modifyCohortAnnotationDefinitionRequest
+          description: Contains the new CohortAnnotationDefinition
+          schema:
+            $ref: "#/definitions/ModifyCohortAnnotationDefinitionRequest"
```

Comment:
Seems like this could take a CohortAnnotationDefinition instead?

Reply:
I could. If i use the CohortAnnotationDefinition then we won't be able to make the cohortAnnotationDefinitionId required.

Reply:
One question is whether we want to support modifying annotation types of existing fields. If we do, there's the problem of what to do with existing annotation values. (Simplest answer: delete them.)

My general preference is to make everything editable... it's annoying to let users create something but not have a way of fixing it subsequently (apart from deleting and recreating.)

Reply:
I think were going to only allow you to modify the name property of the CohortAnnotationDefinition, but not the AnnotationType. So i will remove the AnnotationType from this. So if you create an CohortAnnotationDefinition and it isn't the correct type, you can just delete it and create another one with the correct type.